src/solvers/refinement/string_refinement.cpp

```diff
@@ -900,7 +900,9 @@ decision_proceduret::resultt string_refinementt::dec_solve()
           axioms))
     add_lemma(instance);
 
-  while((loop_bound_--)>0)
+  // loop_bound_ == 0 means no limit; otherwise set max iteration bound:
+  std::size_t refinement_iterations = 0;
+  while(loop_bound_ == 0 || (refinement_iterations++) < loop_bound_)
```

Comment:
0 for no limit is a bit weird. Why not use numeric_limits<size_t>::max? Or do you expect more iterations than that?